test/cases/depsolving.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -euo pipefail
+
+WORKING_DIRECTORY=/usr/libexec/osbuild-composer
+TESTS_PATH=/usr/libexec/osbuild-composer-test
+mkdir --parents /tmp/logs
+LOGS_DIRECTORY=$(mktemp --directory --tmpdir=/tmp/logs)
+
+PASSED_TESTS=()
+FAILED_TESTS=()
+
+TEST_CASES=(
+  "osbuild-dnf-json-tests"
+)
+
+# Print out a nice test divider so we know when tests stop and start.
+test_divider () {
+    printf "%0.s-" {1..78} && echo
+}
+
+# Run a test case and store the result as passed or failed.
+run_test_case () {
+    TEST_NAME=$(basename "$1")
+    echo
+    test_divider
+    echo "🏃🏻 Running test: ${TEST_NAME}"
+    test_divider
+
+    if sudo "${1}" -test.v | tee "${LOGS_DIRECTORY}"/"${TEST_NAME}".log; then
+        PASSED_TESTS+=("$TEST_NAME")
+    else
+        FAILED_TESTS+=("$TEST_NAME")
+    fi
+
+    test_divider
+    echo
+}
+
+# Use repository overrides because RHEL subscription does not allow usage of repositories
+# for multiple architectures (it provides content only for the host's architecture).
+sudo mkdir -p /etc/osbuild-composer/repositories
+sudo cp /usr/share/tests/osbuild-composer/repositories/rhel-8.json /etc/osbuild-composer/repositories/
+sudo cp /usr/share/tests/osbuild-composer/repositories/rhel-84.json /etc/osbuild-composer/repositories/
```

Comment:
I think this should be copied to `rhel-8.json` in the same way we do for other tests. Even when we support 8.3 and 8.4 as separate distros in composer, we only have the one set of repositories defined for RHEL8.

If you re-run the tests on each distro we support you'll also catch any libdnf compat issues, so that might be worth it, even though this test already would catch any issues with the packagelists in our image definitions.